cmdstanpy/utils.py

```diff
@@ -409,6 +409,22 @@ def cxx_toolchain_path(
     return compiler_path, tool_path
 
 
+def rewrite_inf_nan(
```

Comment:
We should also test np.float32 versions if they work as they should.

Reply:
By putting the call toward the end of `write_stan_json`, it is all Python objects at this point anyway, but I will add a test.